src/lattice_reaction_system_diffusion.jl

```diff
@@ -0,0 +1,501 @@
+            J_nzval[val_idx_dst] += get_component_value(rates, e_idx)
+        end
+    end
+
+    return SparseMatrixCSC(lrs.nS * lrs.nC, lrs.nS * lrs.nC, J_colptr, J_rowval, J_nzval)
+end
+
+### JumpProblem ###
+
+# Builds a spatial DiscreteProblem from a Lattice Reaction System.
+
+# Creates a DiscreteProblem from a LatticeReactionSystem.
+function DiffEqBase.DiscreteProblem(lrs::LatticeReactionSystem, u0_in, tspan,
+                                    p_in = DiffEqBase.NullParameters(), args...; kwargs...)
+    u0 = lattice_process_u0(u0_in, ModelingToolkit.getname.(species(lrs)), lrs.nC)
+    pC, pD = lattice_process_p(p_in, Symbol.(compartment_parameters(lrs)),
+                               Symbol.(diffusion_parameters(lrs)), lrs)
+    return DiscreteProblem(lrs.rs, u0, tspan, (pC, pD), args...; kwargs...)
+end
+
+# Builds a spatial JumpProblem from a DiscreteProblem containg a Lattice Reaction System.
+function JumpProcesses.JumpProblem(lrs::LatticeReactionSystem, dprob, aggregator, args...;
+                                   name = nameof(lrs.rs),
+                                   combinatoric_ratelaws = get_combinatoric_ratelaws(lrs.rs),
+                                   checks = false, kwargs...)
+    dprob.p isa Tuple{Vector{Vector{Float64}}, Vector{Vector{Float64}}} ||
+        error("Parameters in input DiscreteProblem is of an unexpected type: $(typeof(dprob.p)). Was a LatticeReactionProblem passed into the DiscreteProblem when it was created?")
+    hopping_constants = make_hopping_constants(dprob, lrs)
+    majumps = make_majumps(dprob, lrs, aggregator, hopping_constants, combinatoric_ratelaws,
+                           checks)
+    ___dprob = DiscreteProblem(reshape(dprob.u0, lrs.nS, lrs.nC), dprob.tspan,
+                               first.(dprob.p[1]))
+    return JumpProblem(___dprob, aggregator, majumps, hopping_constants = hopping_constants,
+                       spatial_system = lrs.lattice, save_positions = (true, false))
+end
+
+# Creates the hopping constants from a discrete problem and a lattice reaction system.
+function make_hopping_constants(dprob::DiscreteProblem, lrs::LatticeReactionSystem)
+    diffusion_rates_dict = Dict(Catalyst.compute_all_diffusion_rates(dprob.p[1], dprob.p[2],
+                                                                     lrs))
+    all_diff_rates = [haskey(diffusion_rates_dict, s) ? diffusion_rates_dict[s] : [0.0]
+                      for s in species(lrs)]
+    if length.(all_diff_rates) == 1
+        return Catalyst.matrix_expand_component_values(all_diff_rates,
+                                                       length(vertices(lrs.lattice)))
+    else
+        hopping_constants = [Vector{Float64}() for i in 1:(lrs.nS), j in 1:(lrs.nC)]
+        for (e_idx, e) in enumerate(edges(lrs.lattice))
+            for s_idx in 1:(lrs.nS)
+                push!(hopping_constants[s_idx, e.src],
```

Comment:
`push!` is slow, because it allocates memory several times. Since we know the length of the array (it's `lrs.nS`), we should pre-allocate it.

Reply:
Could point, will update.

Reply:
I have circumvented using `push!`